proxy/src/control/discovery.rs

```diff
@@ -448,12 +446,19 @@ impl <T: HttpService<ResponseBody = RecvBody>> DestinationSet<T> {
                  authority_for_logging: &FullyQualifiedAuthority,
                  addr: SocketAddr,
                  change: CacheChange) {
+        if let CacheChange::Modification = change {
+            // Skip `Modification` cache change events entirely until metadata
+            // changes occur, and are handled correctly.
+            return;
+        }
         let (update_str, update_constructor): (&'static str, fn(SocketAddr) -> Update) =
             match change {
                 CacheChange::Insertion => ("insert", Update::Insert),
                 CacheChange::Removal => ("remove", Update::Remove),
-                CacheChange::Modification =>
-                    ("change metadata for", Update::ChangeMetadata),
+                CacheChange::Modification => {
+                    // TODO: generate `ChangeMetadata` events.
+                    unreachable!();
```

Comment:
Why isn't this just `return;` so that the `if let` isn't necessary?

Reply:
My mistake --- I misunderstood a `return` there as making _the value of `update_constructor` `return` (an invalid return value for that function), but that's incorrect.